pydantic_ai_slim/pydantic_ai/models/xai.py

```diff
@@ -0,0 +1,1046 @@
+            include.append(chat_pb2.IncludeOption.INCLUDE_OPTION_DOCUMENT_SEARCH_CALL_OUTPUT)
+        if model_settings.get('xai_include_collections_search_outputs'):
+            include.append(chat_pb2.IncludeOption.INCLUDE_OPTION_COLLECTIONS_SEARCH_CALL_OUTPUT)
+        if model_settings.get('xai_include_mcp_outputs'):
+            include.append(chat_pb2.IncludeOption.INCLUDE_OPTION_MCP_CALL_OUTPUT)
+
+        # Create and return chat instance
+        return self._provider.client.chat.create(
+            model=self._model_name,
+            messages=xai_messages,
+            tools=tools_param,
+            tool_choice=tool_choice,
+            response_format=response_format,
+            use_encrypted_content=use_encrypted_content,
+            include=include,
+            **xai_settings,
+        )
+
+    async def request(
+        self,
+        messages: list[ModelMessage],
+        model_settings: ModelSettings | None,
+        model_request_parameters: ModelRequestParameters,
+    ) -> ModelResponse:
+        """Make a request to the xAI model."""
+        check_allow_model_requests()
+        model_settings, model_request_parameters = self.prepare_request(
+            model_settings,
+            model_request_parameters,
+        )
+
+        chat = await self._create_chat(messages, cast(XaiModelSettings, model_settings or {}), model_request_parameters)
+        response = await chat.sample()
+        return self._process_response(response)
+
+    @asynccontextmanager
+    async def request_stream(
+        self,
+        messages: list[ModelMessage],
+        model_settings: ModelSettings | None,
+        model_request_parameters: ModelRequestParameters,
+        run_context: RunContext[Any] | None = None,
+    ) -> AsyncIterator[StreamedResponse]:
+        """Make a streaming request to the xAI model."""
+        check_allow_model_requests()
+        model_settings, model_request_parameters = self.prepare_request(
+            model_settings,
+            model_request_parameters,
+        )
+
+        chat = await self._create_chat(messages, cast(XaiModelSettings, model_settings or {}), model_request_parameters)
+        response_stream = chat.stream()
+        yield await self._process_streamed_response(response_stream, model_request_parameters)
+
+    @staticmethod
+    def get_tool_result_content(content: str) -> dict[str, Any] | str | None:
+        """Extract tool result content from a content string.
+
+        Args:
+            content: The content string (may be JSON or plain text)
+
+        Returns:
+            Tool result content as dict (if JSON), string, or None if no content
+        """
+        if content:
+            try:
+                return json.loads(content)
+            except (json.JSONDecodeError, TypeError):
+                return content
+        return None
+
+    @staticmethod
+    def parse_tool_args(arguments: str) -> dict[str, Any]:
```

Comment:
Please make this a private method

Reply:
Move these to private static module methods since they don't require any state.